File: sets/1/challenges/s1c1.py
import base64

def HexToBase64(hex_input):
  hex_decoded = hex_input.decode("hex")
  base64_encoded = base64.b64encode(hex_decoded)
  return base64_encoded

def main():
  hex_input = "49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d"
  print HexToBase64(hex_input)

if __name__ == "__main__":
  main()

File: sets/1/challenges/s1c1.go
package main

import (
    "encoding/base64"
		"encoding/hex"
    "fmt"
		"log"
)

func main() {
	const input = "49276d206b696c6c696e6720796f757220627261696e206c696b65206120706f69736f6e6f7573206d757368726f6f6d";
	fmt.Printf("input, as hex: %s\n", input)

	decoded, err := hex.DecodeString(input)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Printf("input, decoded: %s\n", decoded)

	encoded := base64.StdEncoding.EncodeToString([]byte(decoded))
	fmt.Printf("encoded: %s\n", encoded)
}
